aten/src/ATen/cpu/vec/vec_base.h

```diff
@@ -308,7 +308,7 @@ struct Vectorized {
   Vectorized<T> real() const {
     // complex_t_real is for SFINAE and clarity. Make sure it is not changed.
     static_assert(std::is_same<complex_t_real, T>::value, "complex_t_real must be T");
-    return map([](T x) { return static_cast<T>(x.real()); });
+    return map([](T x) { return T{x.real(), 0}; });
```

Comment:
Currently `complex<Half>` does not have a constructor which takes only real value. Adding such leads to ambiguous constructor since it matches both real only constructor and the constructor which takes complex<float> as complex<float> has constructor which takes only real value.
Godbole Repro : https://godbolt.org/z/6r1YTbvea

Have instead opted to update this call and other callosities which I believe is semantically same
https://stackoverflow.com/a/32713134/5602957